modular_skyrat/modules/cyborg/code/robot_upgrades.dm

```diff
@@ -244,3 +244,40 @@
 		var/obj/item/inducer/cyborg/inducer = locate() in target_robot.model
 		if (inducer)
 			target_robot.model.remove_module(inducer, TRUE)
+
+// Borg picket sign upgrade
+/obj/item/borg/upgrade/sign
+	name = "Cyborg Multi-purpose advertisement tool upgrade"
+	desc = "A sign unit for cyborgs, allowing them to easily advertise products through an intuitive, wave-able sign."
+	icon_state = "cyborg_upgrade3"
+	require_model = FALSE
+/obj/item/borg/upgrade/sign/action(mob/living/silicon/robot/target_robot, user = usr)
+	. = ..()
+	if(!.)
+		return
+	var/obj/item/picket_sign/cyborg/borgsign = locate() in target_robot.model.modules
+	if(borgsign)
+		to_chat(user, span_warning("This unit is already equipped with an sign module!"))
+		return
+
+	borgsign = new(target_robot.model)
+	target_robot.model.basic_modules += borgsign
+	target_robot.model.add_module(borgsign, FALSE, TRUE)
+
+/obj/item/borg/upgrade/borgsign/deactivate(mob/living/silicon/robot/target_robot, user = usr)
+	. = ..()
+	if (!.)
+		return
+	var/obj/item/picket_sign/cyborg/borgsign = locate() in target_robot.model
+	if (borgsign)
+		target_robot.model.remove_module(borgsign, TRUE)
+
+//Updated proc for the sign, ignores the writing_utensil check in favor of iscyborg(user) This prevent a pesky chat warning.
+/obj/item/picket_sign/cyborg/proc/borgsign_retext(mob/user)
```

Comment:
remove the /proc here

Reply:
It's defined here modularly because the nonmodular proc requires a writing_instrument. This proc does not require that. If the writing_instrument check was left in users would receive "You can't write with !" In their chat every time the action button was used.